services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsRouteTest.java

```diff
@@ -73,4 +77,66 @@ public void directionsRoute_doesContainOptionsAndUuid() throws Exception {
     assertEquals(options, route.routeOptions());
     assertEquals(uuid, route.requestUuid());
   }
+
+  @Test
+  public void directionsRoute_json_withOptionsAndUUID_roundTripping() throws Exception {
+    String json = loadJsonFixture("directions_v5-with-closure_precision_6.json");
+    RouteOptions options = RouteOptions.builder()
+      .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+      .coordinatesList(new ArrayList<Point>() {{
+        add(Point.fromLngLat(1.0, 1.0));
+        add(Point.fromLngLat(2.0, 2.0));
+      }})
+      .accessToken("token")
+      .build();
+    String uuid = "123";
+    DirectionsRoute route = DirectionsRoute.fromJson(json, options, uuid);
+
+    String newRouteJson = route.toJson();
+
+    DirectionsRoute newRoute = DirectionsRoute.fromJson(newRouteJson, "token");
+
+    assertEquals(route, newRoute);
+  }
+
+  @Test
+  public void directionsRoute_json_invalid_with_options() throws Exception {
+    thrown.expect(RuntimeException.class);
+    thrown.expectMessage(
+      "Provided serialized route contains RouteOptions. "
+        + "Use DirectionsRoute#fromJson(json, accessToken) instead.");
+    String json = loadJsonFixture("directions_v5-with-closure_precision_6.json");
+    RouteOptions options = RouteOptions.builder()
+      .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+      .coordinatesList(new ArrayList<Point>() {{
+        add(Point.fromLngLat(1.0, 1.0));
+        add(Point.fromLngLat(2.0, 2.0));
+      }})
+      .accessToken("token")
+      .build();
+    String uuid = "123";
+    DirectionsRoute route = DirectionsRoute.fromJson(json, options, uuid);
+
+    String newRouteJson = route.toJson();
+
+    DirectionsRoute newRoute = DirectionsRoute.fromJson(newRouteJson);
+
+    assertEquals(route, newRoute);
```

Comment:
```suggestion
    DirectionsRoute.fromJson(newRouteJson);
```

Reply:
I don't think I understand this suggestion :thinking: could you explain?

Reply:
It's simple the `assert` is never executed 😅 

What makes the test to pass / throw the exception is `DirectionsRoute.fromJson(newRouteJson);` the rest is unnecessary.

Reply:
lol, good catch :+1: